app/models/patient.rb

```diff
@@ -135,19 +137,25 @@ class Patient
   mongoid_userstamp user_model: 'User'
 
   # Methods
-  def self.pledged_status_summary(num_days = 7)
-    # return pledge totals for patients with appts in the next num_days
-    # TODO move to Pledge class, when implemented?
-    outstanding_pledges = 0
-    sent_total = 0
-    Patient.where(:appointment_date.lte => Date.today + num_days).each do |patient|
-      if patient.pledge_sent
-        sent_total += (patient.fund_pledge || 0)
+  def self.pledged_status_summary(lines = LINES)
+    # return pledge totals for patients with appts in a given week
+    start_of_week = Time.zone.today.beginning_of_week(:monday)
+    plucked_attrs = [:fund_pledge, :pledge_sent, :id, :name, :appointment_date, :fund_pledged_at]
+
+    patients = Patient.in(line: lines)
+                      .where(:fund_pledge.nin => [0, nil, ''])
+                      .where(:fund_pledged_at.gte => start_of_week)
+                      .pluck(*plucked_attrs)
+                      .map { |att| plucked_attrs.zip(att).to_h }
+
+    patients.each_with_object(sent: [], pledged: []) do |patient, summary|
```

Comment:
todo self: lookup this function

Reply:
It's basically `reduce`. I have no idea why they can't just be good with reduce.